### PR TITLE
Add hadolint to pre-commit hooks

### PR DESCRIPTION
**Pull request by aaronmondal**
This hardens the Dockerfile.

- - -
This change is [Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/422)

### CODE REVIEW

File: deployment-examples/docker-compose/Dockerfile [hunk 1]
```diff
@@ -25,21 +25,22 @@ ARG ADDITIONAL_SETUP_WORKER_CMD=
...
-        git \
-        gcc \
-        g++ \
-        python3 \
-        ca-certificates && \
-    npm install -g @bazel/bazelisk
-
+RUN apt-get update && \
+    DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y \
+        npm=8.5.1~ds \
```
Comment (MarcusSorealheis) [inline deployment-examples/docker-compose/Dockerfile]: this is a huge improvement

Comment (MarcusSorealheis) [reply deployment-examples/docker-compose/Dockerfile]: Even if it seems small

**Review by MarcusSorealheis — APPROVED** [review]
I thought about it, and the only issue I see in this PR is that there is no mention of Hadolint in the docker-compose README. Then it dawned me on that we don't have a README for this directory. I've created a new issue [here](https://github.com/TraceMachina/native-link/issues/425)

LGTM

**Review by allada — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/native-link/422)* status: all files reviewed, 2 unresolved discussions (waiting on @aaronmondal)

___
*[`deployment-examples/docker-compose/Dockerfile` line 74 at r2](https://reviewable.io/reviews/TraceMachina/native-link/422#-NkLsp113fbvhD-MM7RT:-NkLsp113fbvhD-MM7RU:b7y5qw4) ([raw file](https://github.com/TraceMachina/native-link/blob/5708dabddd83d2e6de8e96c916f21ad6eebc108e/deployment-examples/docker-compose/Dockerfile#L74)):*
> ```Dockerfile
>         DEBIAN_FRONTEND=noninteractive \
>         apt-get install --no-install-recommends -y \
>             ca-certificates=20230311ubuntu0.22.04.1; \
> ```

nit: Not needed any more, right?

**Review by allada — COMMENTED** [review]
:lgtm:

> Dismissed @MarcusSorealheis from a discussion.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/native-link/422#-:-NkR-6OlARim6kQu2l2l:bnfp4nl)* status: 1 of 2 files reviewed, all discussions resolved (waiting on @aaronmondal)

**Review by aaronmondal — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tracemachina/nativelink/422)* status: <span>:shipit:</span> complete! 1 of 1 LGTMs obtained

___
*[`deployment-examples/docker-compose/Dockerfile` line 74 at r2](https://reviewable.io/reviews/tracemachina/nativelink/422#-NkLsp113fbvhD-MM7RT:-Ns3owZr5I7TAUaNVl1Z:btxmq) ([raw file](https://github.com/tracemachina/nativelink/blob/5708dabddd83d2e6de8e96c916f21ad6eebc108e/deployment-examples/docker-compose/Dockerfile#L74)):*
<details><summary><i>Previously, allada (Nathan (Blaise) Bruer) wrote…</i></summary><blockquote>

nit: Not needed any more, right?
</blockquote></details>

Unfortunately still needed. Will attempt to remove it https://github.com/TraceMachina/nativelink/issues/708

**Pull request merged**